src/main/java/cn/nukkit/utils/Utils.java

```diff
@@ -209,6 +209,9 @@ public static void reverseArray(Object[] data) {
     }
 
     public static int toInt(Object number) {
+        if (number instanceof Double){
+            return Double.valueOf(Math.round((double) number)).intValue();
+        }
         return (int) number;
```

Comment:
What about floats? The original method was more effective because casting a float to a double and then rounding cause no precision to be lost.

Reply:
If you put a Double or Float to (int) something, it will throw error.

You can convert Integer, double, float... to int. But not Double. That’s why I changed it

Reply:
This is probably the best way. It will throw an Arithmetic exception if a double or long is larger than `Integer.MAX_VALUE`
```
    public static int toInt(Number number) {
        return Math.toIntExact(Math.round(number.doubleValue()));
    }
```